pkg/provisioner/ironic/ironic.go

```diff
@@ -50,6 +59,15 @@ const (
 func init() {
 	// NOTE(dhellmann): Use Fprintf() to report errors instead of
 	// logging, because logging is not configured yet in init().
+	authStrategy = os.Getenv("IRONIC_AUTH_STRATEGY")
+	if authStrategy == "" {
+		fmt.Fprintf(os.Stderr, "Cannot start: No IRONIC_AUTH_STRATEGY variable set\n")
+		os.Exit(1)
+	}
```

Comment:
Making this compulsory introduces a dependency, where the deployment (and possibly config map) has to be updated for the BMO to keep working.
I'm not sure this is a good idea, but at the very least we need to update the deploy files in the repo to set it to `noauth` by default.

Reply:
@zaneb the file would only https://github.com/metal3-io/baremetal-operator/blob/master/deploy/default/ironic_bmo_configmap.env ?

Reply:
Ah yep, looks like it is [that simple](https://github.com/metal3-io/baremetal-operator/blob/master/deploy/operator/bmo.yaml#L30-L32) here.

In the downstream MAO, we don't pass a whole config-map (which is in a different format) but [select only the keys we need](https://github.com/openshift/machine-api-operator/blob/master/pkg/operator/baremetal_pod.go#L282-L285), so it will require a simultaneous change (if it requires an addition to the config-map then I suspect there's also a change to the installer required, but I think it's probably safe not to do that since the MAO owns the topology of the deployment).

Reply:
So downstream before we push the BMO changes we would need to update MAO to at least have noauth right?

Reply:
As it stands, yes.